v13 fix

RedSparr0w opened the pull request:
**Please describe the changes this PR makes and why it should be merged:**
Updated to work with Discord.js v13

As far as I can tell from my limited testing, this seems to allow it to work with Discord.js v13+


**Status and versioning classification:**
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

<!--
Please move lines that apply to you out of the comment:
- This PR changes the library's interface (methods or parameters added)
- This PR **only** includes non-code changes, like changes to documentation, README, etc.
-->

RedSparr0w commented:
As far as I could tell everything was working as it should although I haven't used this before so I may have just not noticed some things.

TeoxModding commented:
> As far as I could tell everything was working as it should although I haven't used this before so I may have just not noticed some things.

I think is need to Change the Message.channel.send(`here`, embed) to Message.channel.send({content: `Here`, embeds: [embed]})

That Was One of the Things That Changed About V13
https://discord.js.org/#/docs/main/stable/class/TextChannel?scrollTo=send

Scraayp reviewed (changes requested):
Please include the changes made to discord.js v13 with: 

- https://discord.js.org/#/docs/main/stable/typedef/ChannelType
- https://discordjs.guide/additional-info/changes-in-v13.html#sending-messages-embeds-files-etc
- The rest I will do my self in the v.6.0 pr

RedSparr0w commented:
> > As far as I could tell everything was working as it should although I haven't used this before so I may have just not noticed some things.
> 
> I think is need to Change the Message.channel.send(`here`, embed) to Message.channel.send({content: `Here`, embeds: [embed]})
> 
> That Was One of the Things That Changed About V13
> [discord.js.org/#/docs/main/stable/class/TextChannel?scrollTo=send](https://discord.js.org/#/docs/main/stable/class/TextChannel?scrollTo=send)

I think it only needs the `{ content: 'my message' }` if there is more than just the message AFAIK.
But I've updated it to use content anyway, and `embeds: [embed]`.
Also updated channel type check.
_Should_ all be fine now 🤞

Scraayp commented:
Now please test it. If it's all fine let me know.

Scraayp commented on index.js:
```diff
@@ -54,17 +54,17 @@ const { EventEmitter } = require('events')
  *
  * @property {number} [maxInterval=2000] Amount of time (ms) in which messages are considered spam.
- * @property {number} [maxDuplicatesInterval=2000] Amount of time (ms) in which duplicate messages are considered spam.
+ * @property {number} [maxDuplicatesInterval=10000] Amount of time (ms) in which duplicate messages are considered spam.
```
I would like to keep this as 2000 for now!

Scraayp commented on index.js:
```diff
@@ -140,17 +140,17 @@ class AntiSpamClient extends EventEmitter {
 
 			maxInterval: options.maxInterval || 2000,
-			maxDuplicatesInterval: options.maxDuplicatesInterval || 2000,
+			maxDuplicatesInterval: options.maxDuplicatesInterval || 10000,
```
I would like to keep this as 2000!

Scraayp reviewed (changes requested):
Also please keep this at 2000 for now! Or atleast explain why it should be 10000!

PuneetGopinath commented:
Hello all

A small notice:
All SomethingID has been changed to SomethingId in v13
And when exploring the code I saw it checks if `message.guild.ownerID === message.author.id` where ownerID is no longer a property in djs v13

_If you want me to open a pr, let me know_

Scraayp commented:
I will add it to the 2.6.0 update thanks
